src/can.cpp

```diff
@@ -0,0 +1,92 @@
+#include "cand/can.hpp"
+
+CAN::CAN(std::string name)
+    : m_interfaceName { name }
+{
+
+    int socketCan;
+    struct sockaddr_can addr;
+    struct ifreq ifr;
+
+    if ((socketCan = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
+        throw std::runtime_error("Socket initialization error");
+    }
+
+    strcpy(ifr.ifr_name, m_interfaceName.c_str());
```

Comment:
O strcpy já resolve o problema, mas sugiro o uso doo strncpy por ser mais seguro. Acredito ser uma melhor prática.

```suggestion
    strncpy(ifr.ifr_name, m_interfaceName.c_str(), 32);
```

Reply:
em vez de colocar o tamanho seria melhor colocar ```sizeof(ifr.ifr_name)```

Reply:
Boa ideia, vou editar a sugestão aqui.